src/components/decimal/decimal.test.js

```diff
@@ -0,0 +1,302 @@
+    });
+
+    it("should render Decimal with readOnly prop", () => {
+      // and mount the story using @cypress/react library
+      mount(<Decimal readOnly />);
+
+      const inputValue = "test";
+
+      // then run our tests
+      cy.get(decimalComponent)
+        .type(inputValue)
+        .find("input")
+        .blur({ force: true });
+      cy.get(decimalComponent)
+        .find("input")
+        .should("not.have.value", inputValue)
+        .and("have.attr", "readOnly");
+    });
+  });
+
+  describe("check Decimal input", () => {
+    const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+
+    it("check label renders proper specific values", () => {
+      // and mount the story using @cypress/react library
+      testData.forEach((element) => {
+        mount(<Decimal fieldHelp={element} />);
+
+        // then run our tests
+        fieldHelpPreview().should("have.text", element);
+      });
+    });
+
+    it("check fieldHelp renders proper specific values", () => {
+      // and mount the story using @cypress/react library
+      testData.forEach((element) => {
+        mount(<Decimal label={element} />);
+
+        // then run our tests
+        getDataElementByValue("label").should("have.text", element);
+      });
+    });
+
+    it("check tooltip renders proper specific values", () => {
+      // and mount the story using @cypress/react library
+      testData.forEach((element) => {
+        mount(<Decimal label="Label" labelHelp={element} />);
+
+        // then run our tests
+        getDataElementByValue("question").trigger("mouseover");
+        tooltipPreview().should("have.text", element);
+      });
+    });
+
+    it("check Decimal component accepts specific values", () => {
+      // and mount the story using @cypress/react library
+      mount(<Decimal />);
+
+      testData.forEach((element) => {
+        // then run our tests
+        cy.get(decimalComponent)
+          .type(element)
+          .find("input")
+          .blur({ force: true });
+        cy.get(decimalComponent)
+          .invoke("val")
+          .then(($el) => {
+            for (let number = 0; number < $el.length; number++) {
+              expect(
+                $el.replace(/(\s)|(&nbsp;)|(\u00a0)/g, " ").charCodeAt(number)
+              ).to.equals(element.charCodeAt(number));
+            }
+          });
+      });
+    });
+
+    it("check Decimal component accepts white spaces", () => {
+      // and mount the story using @cypress/react library
+      mount(<Decimal />);
+
+      // then run our tests
+      cy.get(decimalComponent).type("   ").find("input").blur({ force: true });
+      cy.get(decimalComponent)
+        .find("input")
+        .should("have.attr", "value", "   ");
+    });
+  });
+
+  describe("check events for Decimal component", () => {
+    it("should return onChange event", () => {
+      // use cy.stub for mock the onChange callback
+      const callback = cy.stub().as("eventCallback");
```

Comment:
Why does this have a `eventCallback`, the other stubs do not.

Reply:
just a test to make possible to use
`cy.get(@ eventCallback)` - will delete this :)